grr/server/grr_response_server/gui/ui/components/client_add_label_dialog/client_add_label_dialog.ng.html

```diff
@@ -0,0 +1,24 @@
+<div mat-dialog-title class="title">Adding a label</div>
+<div mat-dialog-content class="content">
+  <p>You can add a new label or select from existing labels.</p>
+  <mat-form-field class="label-field" appearance="outline">
+    <mat-label>Start typing label name</mat-label>
+    <input matInput
+      cdkFocusInitial
+      (keyup.enter)="onAddClick()"
+      [formControl]="newLabelInputControl"
+      [matAutocomplete]="auto">
+
+    <mat-autocomplete #auto="matAutocomplete">
+      <mat-option *ngFor="let option of filteredCurrentLabels | async" [value]="option">
```

Comment:
Is this autocomplete only suggesting already pre-existing labels of the client?

Reply:
Exactly!

Reply:
We want to suggest all labels in the autocomplete, not only labels that are already assigned to the client.

Reply:
All labels from all clients? Is there an API for this?

Reply:
Yes, use `ApiListClientsLabelsHandler` via `GET /api/v2/clients/labels`.

To clarify: Clients and Labels have a many-to-many relationship. Any Client can have `[0,n]` labels assigned and any label be assigned to many clients. As an implementation detail, if a label is no longer assigned to any client, it is also deleted from the database, but that does not need to concern you here.

When adding a label to a Client, the user can type in an arbitrary string to be used as label. The autocomplete should suggest any pre-existing labels, not only labels that are already assigned to the current client.